shotover-proxy/src/message/mod.rs

```diff
@@ -232,6 +241,56 @@ impl Message {
         self.invalidate_cache();
     }
 
+    /// Get metadata for this `Message`
+    fn metadata(&mut self) -> Result<Metadata> {
+        let inner = self.inner.take().unwrap();
```

Comment:
take here is quite scary.
This happens to work because we always destroy the message after calling metadata but we should be able to implement metadata without taking (and leaving as None) the self.inner

Reply:
How can we parse the bytes without cloning them for the `MessageInner::RawBytes` variant?

Reply:
Just pass in an &[u8]

Reply:
this works for me:
![image](https://user-images.githubusercontent.com/5120858/156469939-ec020072-7a25-43a7-aed4-0b637c398990.png)